middlewares/multer/avatarUpload.js

```diff
@@ -0,0 +1,35 @@
+const { ErrorObject } = require("../../helpers/error");
+const multer = require("multer");
+const ds = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null, "public/avatars");
+  },
+  filename: (req, file, cb) => {
+    req.body.avatar = `http://localhost:${process.env.PORT || 3000}/avatars/${
+      file.originalname
+    }`;
+    cb(null, file.originalname);
+  },
+});
+const multerConfig = multer({
+  fileFilter: (req, file, cb) => {
+    if (
+      file.mimetype === "image/png" ||
+      file.mimetype === "image/jpg" ||
+      file.mimetype === "image/svg" ||
+      file.mimetype === "image/webp"
+    )
+      return cb(null, true);
+
+    return cb(new ErrorObject("Invalid avatar file extension.", 422));
+  },
+  storage: ds,
+});
+const uploadAvatar = multerConfig.single("avatar");
+
+module.exports = (req, res, next) => {
+  uploadAvatar(req, res, (error) => {
+    if (error) return res.status(error.statusCode).json(error);
```

Comment:
error in this line: 

whitout code:

![image](https://user-images.githubusercontent.com/84211273/201226315-81c39b5b-23c6-4fe7-a16f-54d2ea779df9.png)

![image](https://user-images.githubusercontent.com/84211273/201226330-a6a20ac2-078e-438f-af14-f560a3a2c1d6.png)

whit code:

![image](https://user-images.githubusercontent.com/84211273/201226350-353ad4bd-ccc4-440b-86b0-a69eef82efe4.png)